core/src/avm1/globals/error.rs

```diff
@@ -0,0 +1,66 @@
+//! Error object
+
+use crate::avm1::activation::Activation;
+use crate::avm1::error::Error;
+use crate::avm1::property::Attribute::*;
+use crate::avm1::{Object, ScriptObject, TObject, UpdateContext, Value};
+use gc_arena::MutationContext;
+
+pub fn constructor<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let message = match args.get(0).cloned() {
+        Some(Value::String(s)) => s,
+        Some(v) => v.coerce_to_string(activation, context)?.to_string(),
+        _ => String::new(),
+    };
+
+    this.set("message", message.into(), activation, context)?;
+
+    // Default name for the object is "Error", which isn't provided by constructor
+    this.set("name", "Error".into(), activation, context)?;
```

Comment:
These defaults come from the prototype so shouldn't be set here.

Reply:
Ah, do you mean using `object.add_property` in `create_proto` instead?

Reply:
From what I can tell they are just standard values, so something like this in `create_proto`:
`error.define_value(gc_context, "message", "Error".into(), EnumSet::empty());`

By default, if you set no message, `error.message` just falls back to the prototype value and returns "Error".
But if you set `error.message = "foo";`, this creates a new value on the Error itself that takes precedence over the prototype value.